### PR TITLE
[FrameworkBundle][DX] Add option to specify additional translation loading paths

### PR DESCRIPTION
**Pull request by Seldaek**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #971 |
| License | MIT |
| Doc PR | TODO |

Thanks to this PR, next time I want to just do:

```
    translator:
        fallback: "%locale%"
        paths: ['%kernel.root_dir%/../vendor/internal/package/translations']
```

I will be able to, and won't have to dig in docs, start building a translation loader before I realize it's not what I want, dig in the framework ext, and then have to write a compiler pass to inject my file to the translator resource paths.

### CODE REVIEW

**Comment by aitboudad** [comment]
@Seldaek see #14399

**Comment by Seldaek** [comment]
Oh fail I did not see that.. I'd argue mine wins for testing it works and not just testing it fails though ;)

Also my version adds the paths at the best location IMO which is over the symfony translations but under the application ones. Anyway I'll leave open and leave our dear @symfony/deciders decide because I spent enough time on this as it is :)

**Comment by aitboudad** [comment]
:smile: well I closed my own PR, what do you think about adding [priority](https://github.com/symfony/symfony/pull/14399#issuecomment-94318105) ?

**Comment by weaverryan** [comment]
I definitely support this idea. But, even further, (and please don't let this get off-topic - this comment is off-topic enough), I wish resource paths were registered in a way more like how this change allows - i.e. bundles register their own translation paths, instead of having the core look at every bundle and register paths *for* them. I think that adds to the "magic" of Symfony - a lot is automatic when you use a bundle. I experimented with an approach for templates in https://github.com/symfony/symfony/pull/13928/files#diff-40488c4aeae121d3032bbe13e2f5801eR26 where bundles register template paths, instead in core.

Thanks!

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -583,6 +583,13 @@ private function addTranslatorSection(ArrayNodeDefinition $rootNode)
                         ->end()
                         ->booleanNode('logging')->defaultValue($this->debug)->end()
                     ->end()
+                    ->fixXmlConfig('path')
+                    ->children()
+                        ->arrayNode('paths')
+                            ->prototype('scalar')->end()
+                            ->defaultValue(array())
+                        ->end()
+                    ->end()
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: What about merging this node and the `fallbacks` node into one single `children()` block?

Comment (Seldaek) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: Yeah that's my bad I'll fix.

**Comment by Seldaek** [comment]
@aitboudad not sure if we need such a complex system that priority is necessary or not. I mean the need doesn't seem to have come up until now so I doubt it's a real problem. Usually you just want to override stuff from third parties in your app and that's it.

@weaverryan sounds like a good improvement for another PR yeah but I know I won't have time to pull it off :)

**Comment by xabbuh** [comment]
@weaverryan @Seldaek Sounds like a use case for Puli. :) /cc @webmozart

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -582,6 +583,10 @@ private function addTranslatorSection(ArrayNodeDefinition $rootNode)
                             ->defaultValue(array('en'))
                         ->end()
                         ->booleanNode('logging')->defaultValue($this->debug)->end()
+                        ->arrayNode('paths')
+                            ->prototype('scalar')->end()
+                            ->defaultValue(array())
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: the defaultValue is useless here. It is already the case for a prototyped node

Comment (Seldaek) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: Right, fixing

**Comment by weaverryan** [comment]
@xabbuh I agree (at least in theory - I've talked with Bernhard on Skype about Puli), but it's a very complex problem, especially when you think about integrating it with Symfony and keeping BC as much as possible. Right now, I'm interested in what our  ideal API for setting paths to translations, views, etc would look like. When that's more clear, we should see if Puli can fill in for a lot of the work that would need to make this happen.

**Comment by fabpot** [comment]
Ready to be merged?

**Comment by Seldaek** [comment]
To me yes. It doesn't prevent further improvements as were discussed above.

**Comment by aitboudad** [comment]
:+1:

**Comment by weaverryan** [comment]
Yep - my comments should not stop merging at all :+1: from me - it looks very simple.

**Comment by aitboudad** [comment]
Thank you @Seldaek.

**Comment by aitboudad** [comment]
Doc issue https://github.com/symfony/symfony-docs/issues/5236

**Pull request closed without merging**